docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+ * Be careful with CSS preprocessor features. Unlike Trello, we use SASS but the principle
+   of using 'vanilla' CSS and being careful with preprocessor features still applies.
+   See the note about `@extends` below
+ * Use naming conventions to indicate which classes are purely presentational and which are referenced in code
+
+## File Structure
+
+The SASS for Hypothesis projects is organized into several parts:
+
+ * A 'root' file which imports all of the components used in the site, structured like this:
+
+```scss
+// Variables and mixins
+@import 'partials/base';
+
+// Reset default browser styles
+@import 'partials/reset';
+// Basic universal styling for elements (eg. links)
+@import 'partials/elements';
+
+// Components
+@import 'partials/input-field';
+@import 'partials/fancy-modal';
+@import 'partials/a-widget';
+```
+
+ * A 'base' partial which imports all variables and mixins used by other components
+ * A 'variables' partial which defines the colors, typography, grid spacing, animation durations and other foundational elements of the design system
+ * Component partials which define the styles for re-usable components, including the different variations of a component and media queries for responsive layout. These may assume
+   the existence of other variables or mixins but should not import anything themselves.
+
+## Selectors
+
+ * Use simple non-nested class selectors for the majority of styling
+ * Descendant selectors should be avoided. They [are slow](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Writing_efficient_CSS#Avoid_the_descendant_selector.21) and more importantly, make it easy to create unintended results due to naming clashes or [specificity conflicts](http://css.maxdesign.com.au/selectutorial/advanced_conflict.htm).
+
+## Naming
+
+ * Use [BEM-style](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/) naming for components:
+
+ ```js
+ .component { ... }
+ .component__element { ... }
+ .component__element--modifier { ... }
+ ```
+
+ Where 'component' is a re-usable building block (eg. an input field),
+ 'element' is some part of the component which is not reusable on its own, and
+ '--modifier' represents some static variation of a component (eg.
+ `btn--danger` for a button which performs some destructive action)
+ * Use `is-$state` names for state classes (eg. `is-active`) that are dynamically added or removed from JS code
+ * Use names with a `js-` prefix for classes that are used by JS to get a reference to particular DOM elements. These classes should not appear in stylesheets
+ * Except for tests, only styles with `js-` and `is-` prefixes should be referenced by JavaScript code
+ * Use a `u-` prefix for utility classes. eg. A class for hiding DOM elements would be called `u-hidden`
+
+## SASS Features
+
+ * Use variables and mixins
+ * [Avoid @extends](https://www.sitepoint.com/avoid-sass-extend/). Use mixins instead
+ * Use nesting sparingly
```

Comment:
Might be worth clarifying what this means, because "nesting" could refer to several things. "Use nesting of rules sparingly -- overly nested SCSS can impede understanding."